### PR TITLE
Add PerconaXtraDB OpsReq and Autoscaler APIs

### PR DESCRIPTION
**Pull request by spectro30**
Signed-off-by: Md. Alif Biswas <alif@appscode.com>

### CODE REVIEW

File: apis/ops/v1alpha1/perconaxtradb_ops_types.go [hunk 1]
```diff
@@ -64,43 +66,90 @@ type PerconaXtraDBOpsRequestSpec struct {
...
+	//	applyConfig:
+	//		file-name.cnf: |
+	//			[mysqld]
+	//			key1: value1
+	//			key2: value2
+	// +optional
+	ApplyConfig map[string]string `json:"applyConfig,omitempty"`
+}
 
 type PerconaXtraDBCustomConfiguration struct {
```
Comment (heheh13) [inline apis/ops/v1alpha1/perconaxtradb_ops_types.go:127]: didn't find any usage

**Review by heheh13 — CHANGES_REQUESTED** [review]

File: apis/ops/v1alpha1/perconaxtradb_ops_types.go (hunk 1 above)
Comment (spectro30) [reply apis/ops/v1alpha1/perconaxtradb_ops_types.go:127]: Here is the usage:
https://github.com/kubedb/ops-manager/blob/add-px-ops/pkg/controller/perconaxtradb/reconfigure.go#L68-L164

**Review by heheh13 — APPROVED** [review]
LGTM!

File: apis/ops/v1alpha1/perconaxtradb_ops_types.go [hunk 2]
```diff
@@ -64,41 +66,83 @@ type PerconaXtraDBOpsRequestSpec struct {
...
+	// +optional
+	RequireSSL *bool `json:"requireSSL,omitempty"`
+}
 
-type PerconaXtraDBCustomConfiguration struct {
-	ConfigMap *core.LocalObjectReference `json:"configMap,omitempty"`
-	Data      map[string]string          `json:"data,omitempty"`
-	Remove    bool                       `json:"remove,omitempty"`
+// PerconaXtraDBOpsRequestStatus is the status for PerconaXtraDBOpsRequest
+type PerconaXtraDBOpsRequestStatus struct {
```
Comment (tamalsaha) [inline apis/ops/v1alpha1/perconaxtradb_ops_types.go:137]: Don't need this.

File: apis/autoscaling/v1alpha1/perconaxtradb_types.go [hunk 3]
```diff
@@ -42,118 +41,44 @@ const (
...
-	// If not set, the default value MaxPolicySelect is used.
-	// +optional
-	SelectPolicy *v2beta2.ScalingPolicySelect `json:"selectPolicy,omitempty"`
-	// policies is a list of potential scaling polices which can be used during scaling.
-	// At least one policy must be specified, otherwise the PerconaXtraDBScalingRules will be discarded as invalid
-	// +optional
-	Policies []PerconaXtraDBScalingPolicy `json:"policies,omitempty"`
+type PerconaXtraDBComputeAutoscalerSpec struct {
+	PerconaXtraDB    *ComputeAutoscalerSpec `json:"perconaxtradb,omitempty"`
+	DisableScaleDown bool                   `json:"disableScaleDown,omitempty"`
```
Comment (tamalsaha) [inline apis/autoscaling/v1alpha1/perconaxtradb_types.go:67]: Remove `DisableScaleDown `

File: apis/ops/v1alpha1/perconaxtradb_ops_types.go (hunk 2 above)
Comment (spectro30) [reply apis/ops/v1alpha1/perconaxtradb_ops_types.go]: removed

File: apis/autoscaling/v1alpha1/perconaxtradb_types.go (hunk 3 above)
Comment (spectro30) [reply apis/autoscaling/v1alpha1/perconaxtradb_types.go]: removed

**Review by tamalsaha — APPROVED** [review]

**Pull request merged**